internal/internal_task_pollers.go

```diff
@@ -256,9 +252,19 @@ func (bp *basePoller) doPoll(
 	ctx, cancel, _ := newChannelContext(context.Background(), featureFlags, chanTimeout(pollTaskServiceTimeOut))
 
 	go func() {
+		defer close(doneC)
+		defer func() {
+			if p := recover(); p != nil {
+				bp.metricsScope.Counter(metrics.WorkerPanicCounter).Inc(1)
+				st := getStackTraceRaw("base poller [panic]:", 7, 0)
```

Comment:
can you do a soft validation on the stackTrace to make sure that omitting 7 lines is actually trimming right amount of lines from the stack and leaving the correct caller as the top of the stack?

Reply:
I just verified it, apparently 7 is the magic number regardless of the call site. Every use of this function passes the same value.